### PR TITLE
Add distributed PyTorch training example on CIFAR-10

### LINKED ISSUE #3: Add distributed training pytorch notebook
**What notebook?**
What would you like to see in an example notebook? 
Showcase pytorch DDP distributed training

**Additional context**
Add any other context here.

### PR DESCRIPTION
**Pull request by mx-iao**
# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines
- [x] ran `python readme.py` after making changes (per contributing guidelines)
- [x] added required testing (per contributing guidelines)

## Changes

-

fixes #3

### CODE REVIEW

**Review by aminsaied — APPROVED** [review]

**Comment by lostmygithubaccount** [comment]
training takes a bit long (it's almost finished) - maybe reduce # of epochs

**Review by lostmygithubaccount — APPROVED** [review]

File: workflows/train/pytorch/cifar-distributed/src/train.py
Comment (dkmiller) [inline workflows/train/pytorch/cifar-distributed/src/train.py:112]: I suggest:

1. Put the logic to create the appropriate argument parser in a standalone method.
2. Call `parser.parse_args()` **outside** the main method (in the `if __name__ == "__main__"`) block, and pass the `args` into `main` as a parameter.

**Review by dkmiller — CHANGES_REQUESTED** [review]

**Review by aminsaied — APPROVED** [review]

File: workflows/train/pytorch/cifar-distributed/src/train.py
Comment (lostmygithubaccount) [reply workflows/train/pytorch/cifar-distributed/src/train.py:112]: @dkmiller would you like to contribute this change?

Comment (lostmygithubaccount) [reply workflows/train/pytorch/cifar-distributed/src/train.py]: fixed

**Comment by lostmygithubaccount** [comment]
changed this repo to no longer dismiss reviews on new commits - will admin merge this once tests pass given it's been signed off

File: workflows/train/pytorch/cifar-distributed/src/train.py
Comment (dkmiller) [inline workflows/train/pytorch/cifar-distributed/src/train.py:238]: Don't rely on a global `args` variable: explicitly pass it in to the main method.

**Review by dkmiller — CHANGES_REQUESTED** [review]

File: workflows/train/pytorch/cifar-distributed/src/train.py
Comment (lostmygithubaccount) [reply workflows/train/pytorch/cifar-distributed/src/train.py:238]: whoops good catch

Comment (lostmygithubaccount) [reply workflows/train/pytorch/cifar-distributed/src/train.py]: resolved in latest commit

**Review by dkmiller — APPROVED** [review]

**Pull request merged**
